guide/src/guide/new-cell.md

```diff
@@ -1 +1,221 @@
-# Write a New Cell (TODO)
+# Write a New Cell
```

Comment:
We should tell the readers why and how they can implement a new cell for the whole guide. That is, why we should implement certain traits and how it affects the implementation, and what we can do and what we cannot do inside the trait.